cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateTestExecutableTask.java

```diff
@@ -0,0 +1,232 @@
+
+    public CreateTestExecutableTask(PrintStream out, String output, RunTestsTask runTestsTask) {
+        super(out, output);
+        this.runTestsTask = runTestsTask;   //to use the getCmdArgs() method
+    }
+
+    @Override
+    public void execute(Project project) {
+        this.out.println();
+
+        this.currentDir = Paths.get(System.getProperty(USER_DIR));
+        Target target = getTarget(project);
+
+        try {
+            PackageCompilation pkgCompilation = project.currentPackage().getCompilation();
+            JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(pkgCompilation, JvmTarget.JAVA_17);
+            JarResolver jarResolver = jBallerinaBackend.jarResolver();
+
+            List<Diagnostic> diagnostics = new ArrayList<>();
+            Path testCachePath = target.getTestsCachePath();
+
+            long start = 0;
+            if (project.buildOptions().dumpBuildTime()) {
+                start = System.currentTimeMillis();
+            }
+
+            HashSet<JarLibrary> testExecDependencies = new HashSet<>();
+            Map<String, TestSuite> testSuiteMap = new HashMap<>();
+
+            //write the test suite json that is used to execute the tests
+            boolean status = createTestSuiteForCloudArtifacts(project, jBallerinaBackend, target, testSuiteMap);
+
+            //get all the dependencies required for test execution for each module
+            for (ModuleDescriptor moduleDescriptor :
+                    project.currentPackage().moduleDependencyGraph().toTopologicallySortedList()) {
+
+                Module module = project.currentPackage().module(moduleDescriptor.name());
+                testExecDependencies.addAll(jarResolver
+                        .getJarFilePathsRequiredForTestExecution(module.moduleName())
+                );
+            }
+
+            if (status) {
+                Path testExecutablePath = getTestExecutableBasePath(target).resolve(
+                        project.currentPackage().packageName().toString() +
+                                ProjectConstants.TEST_UBER_JAR_SUFFIX +
+                                ProjectConstants.BLANG_COMPILED_JAR_EXT);
+
+                //write the cmd args to a file, so it can be read on c2c side
+                writeCmdArgsToFile(testExecutablePath.getParent(), target, TestUtils.getJsonFilePath(testCachePath));
+
+                List<Path> moduleJarPaths = TestUtils.getModuleJarPaths(jBallerinaBackend, project.currentPackage());
+
+                List<String> excludingClassPaths = new ArrayList<>();
+
+                for (Path moduleJarPath : moduleJarPaths) {
+                    ZipFile zipFile = new ZipFile(moduleJarPath.toFile());
+
+                    zipFile.stream().forEach(entry -> {
+                        if (entry.getName().endsWith(".class")) {
+                            excludingClassPaths.add(entry.getName().replace("/", ".")
+                                    .replace(".class", ""));
```

Comment:
Can use constants here. Check other places as well.

Reply:
Done in [40229a4](https://github.com/ballerina-platform/ballerina-lang/pull/42183/commits/40229a40c6e8385fc2d56f004746b7a5494411f1) commit.